src/sharding/ShardClientUtil.js

```diff
@@ -127,29 +127,29 @@ class ShardClientUtil {
 
   /**
    * Evaluates a script or function on all shards, or a given shard, in the context of the {@link Client}s.
-   * @param {string|Function} script JavaScript to run on each shard
-   * @param {number} [shard] Shard to run script on, all if undefined
+   * @param {Function} script JavaScript to run on each shard
+   * @param {BroadcastEvalOptions} [options={}] The options for the broadcast
    * @returns {Promise<*>|Promise<Array<*>>} Results of the script execution
    * @example
-   * client.shard.broadcastEval('this.guilds.cache.size')
+   * client.shard.broadcastEval(client => client.guilds.cache.size)
    *   .then(results => console.log(`${results.reduce((prev, val) => prev + val, 0)} total guilds`))
    *   .catch(console.error);
    * @see {@link ShardingManager#broadcastEval}
    */
-  broadcastEval(script, shard) {
+  broadcastEval(script, options = {}) {
```

Comment:
I think it would be beneficial to validate that `script` is actually a function.
Otherwise you get some confusing / cryptic errors when attempting to pass strings like before.

Reply:
Should we also check for whether or not the function's body is native code? Such as `Number.isNaN` or `cb.bind(null)` both stringify to `function NameOfFunction() { [native code] }`.